### PR TITLE
Make Production Place repeatable and facetable

### PR DESCRIPTION
**Pull request by philippconzett**
**What this PR does / why we need it**:
Data contained in a dataset may have been produced at multiple places. Making the field Production Place repeatable will make it possible to reflect this fact in the dataset metadata. Making the field facetable will allow us to customize Dataverse collections more appropriately.

**Which issue(s) this PR closes**:
None, to my knowledge.

### CODE REVIEW

**Comment by scolapasta** [comment]
@philippconzett just to be sure this is also changing series, is that intended?

**Comment by philippconzett** [comment]
@scolapasta Yes, that's right. I tried to split this into two PRs, but I don't know how to do that :-/

**Comment by philippconzett** [comment]
@djbrooke Could we get this PR merged before the next release?

**Comment by djbrooke** [comment]
Hi @philippconzett - there's the change on line 69, to series, that I thought you did not want to include in this PR:

https://github.com/IQSS/dataverse/pull/7810/files#diff-9a87ef9131a9320671e94ee2b06991ee64d8ae37ddad2bff052215c55e2ddb10R69

Let me know if you'd like me to work on this. I can make changes to the PR so that only your desired changes are included.

**Comment by philippconzett** [comment]
Hi @djbrooke. Actually, I wanted Series to be made repeatable. I probably should have created this as a separate PR, but I didn't know how to remove it from this one. If it's OK, please merge the change together with the Production Place change. Thanks!

**Comment by jggautier** [comment]
Hi @philippconzett. In your datasets' metadata, is each production place associated with the metadata entered in Producer fields? I'm wondering if it would make sense to make Production Place a child field of the Producer fields.

I know this isn't how the two elements, Production Place and Producer, are organized in DDI Codebook 2 and I think that's why I'm the Citation metadata block they're treated two separate fields.

**Comment by djbrooke** [comment]
Thanks @philippconzett, I've updated the title to reflect that you want both of these changes. 

Thanks @jggautier for jumping in!

**Comment by philippconzett** [comment]
@djbrooke Thanks for changing the title.
@jggautier Thanks for raising the issue. I'm not sure whether Production Place should be a sub-field of Producer. Production Place has the following description: "The location where the data collection and any other related materials were produced." I think for many scientific fields where data are collected and sometimes also processed offsite/outside campus, Production Place is not identical with the location of the research organization. Also, one dataset may have been produced at different locations. So, I think we should keep Production Place and Producer separate, but still make Production Place repeatable. Thanks.

**Comment by jggautier** [comment]
Thanks @philippconzett. That makes sense. This makes me think that "location" in the description doesn't always mean a geographical location. For some datasets, people may need to include the name of a place, like the name of an organization.

@djbrooke you mentioned yesterday that you might have questions?

**Comment by djbrooke** [comment]
@jggautier thanks, no questions at the moment! I'll update from dev and add a release note about updating citation.csv.

**Comment by djbrooke** [comment]
OK - all set. I will keep this assigned to me because I'll need to update from develop once we cut 5.8.

**Comment by kcondon** [comment]
@philippconzett Please update release notes to add reminder about updating the solr schema: https://guides.dataverse.org/en/5.8/admin/metadatacustomization.html#updating-the-solr-schema

Also, unsure whether this needs a reindex?

**Comment by djbrooke** [comment]
Thanks @kcondon 

@philippconzett we'll take care of the remaining release notes updates on this side. Thanks!

File: conf/solr/8.8.1/schema.xml [hunk 1]
```diff
@@ -359,7 +359,7 @@
     <field name="responseRate" type="text_en" multiValued="false" stored="true" indexed="true"/>
     <field name="samplingErrorEstimates" type="text_en" multiValued="false" stored="true" indexed="true"/>
     <field name="samplingProcedure" type="text_en" multiValued="false" stored="true" indexed="true"/>
-    <field name="series" type="text_en" multiValued="false" stored="true" indexed="true"/>
+    <field name="series" type="text_en" multiValued="true" stored="true" indexed="true"/>
```
Comment (pdurbin) [inline conf/solr/8.8.1/schema.xml:362]: I think we want the subfields of "series" to be multivalued as well. Please check out this diff:

```
-    <field name="series" type="text_en" multiValued="false" stored="true" indexed="true"/>
-    <field name="seriesInformation" type="text_en" multiValued="false" stored="true" indexed="true"/>
-    <field name="seriesName" type="text_en" multiValued="false" stored="true" indexed="true"/>
+    <field name="series" type="text_en" multiValued="true" stored="true" indexed="true"/>
+    <field name="seriesInformation" type="text_en" multiValued="true" stored="true" indexed="true"/>
+    <field name="seriesName" type="text_en" multiValued="true" stored="true" indexed="true"/>
```

**Review by pdurbin — CHANGES_REQUESTED** [review]
The "series" subfields should be updated in schema.xml. We plan to give more guidance about this in #8248.

**Comment by pdurbin** [comment]
When I load the TSV in this pull request and do the manual process I used to do I get a bigger "diff" below than I expect.

Here's the resulting schema.xml (renamed so I can upload it): [schema.xml.txt](https://github.com/IQSS/dataverse/files/7572896/schema.xml.txt)

The diff:

```
diff --git a/conf/solr/8.8.1/schema.xml b/conf/solr/8.8.1/schema.xml
index 1718962bfb..0a4e6f3d96 100644
--- a/conf/solr/8.8.1/schema.xml
+++ b/conf/solr/8.8.1/schema.xml
@@ -342,7 +342,7 @@
     <field name="producerName" type="text_en" multiValued="true" stored="true" indexed="true"/>
     <field name="producerURL" type="text_en" multiValued="true" stored="true" indexed="true"/>
     <field name="productionDate" type="text_en" multiValued="false" stored="true" indexed="true"/>
-    <field name="productionPlace" type="text_en" multiValued="false" stored="true" indexed="true"/>
+    <field name="productionPlace" type="text_en" multiValued="true" stored="true" indexed="true"/>
     <field name="publication" type="text_en" multiValued="true" stored="true" indexed="true"/>
     <field name="publicationCitation" type="text_en" multiValued="true" stored="true" indexed="true"/>
     <field name="publicationIDNumber" type="text_en" multiValued="true" stored="true" indexed="true"/>
@@ -359,9 +359,9 @@
     <field name="responseRate" type="text_en" multiValued="false" stored="true" indexed="true"/>
     <field name="samplingErrorEstimates" type="text_en" multiValued="false" stored="true" indexed="true"/>
     <field name="samplingProcedure" type="text_en" multiValued="false" stored="true" indexed="true"/>
-    <field name="series" type="text_en" multiValued="false" stored="true" indexed="true"/>
-    <field name="seriesInformation" type="text_en" multiValued="false" stored="true" indexed="true"/>
-    <field name="seriesName" type="text_en" multiValued="false" stored="true" indexed="true"/>
+    <field name="series" type="text_en" multiValued="true" stored="true" indexed="true"/>
+    <field name="seriesInformation" type="text_en" multiValued="true" stored="true" indexed="true"/>
+    <field name="seriesName" type="text_en" multiValued="true" stored="true" indexed="true"/>
     <field name="socialScienceNotes" type="text_en" multiValued="false" stored="true" indexed="true"/>
     <field name="socialScienceNotesSubject" type="text_en" multiValued="false" stored="true" indexed="true"/>
     <field name="socialScienceNotesText" type="text_en" multiValued="false" stored="true" indexed="true"/>
@@ -376,13 +376,13 @@
     <field name="studyAssayOrganism" type="text_en" multiValued="true" stored="true" indexed="true"/>
     <field name="studyAssayOtherMeasurmentType" type="text_en" multiValued="true" stored="true" indexed="true"/>
     <field name="studyAssayOtherOrganism" type="text_en" multiValued="true" stored="true" indexed="true"/>
-    <field name="studyAssayPlatform" type="text_en" multiValued="true" stored="true" indexed="true"/>
     <field name="studyAssayOtherPlatform" type="text_en" multiValued="true" stored="true" indexed="true"/>
-    <field name="studyAssayTechnologyType" type="text_en" multiValued="true" stored="true" indexed="true"/>
     <field name="studyAssayOtherTechnologyType" type="text_en" multiValued="true" stored="true" indexed="true"/>
+    <field name="studyAssayPlatform" type="text_en" multiValued="true" stored="true" indexed="true"/>
+    <field name="studyAssayTechnologyType" type="text_en" multiValued="true" stored="true" indexed="true"/>
     <field name="studyDesignType" type="text_en" multiValued="true" stored="true" indexed="true"/>
-    <field name="studyOtherDesignType" type="text_en" multiValued="true" stored="true" indexed="true"/>
     <field name="studyFactorType" type="text_en" multiValued="true" stored="true" indexed="true"/>
+    <field name="studyOtherDesignType" type="text_en" multiValued="true" stored="true" indexed="true"/>
     <field name="studyOtherFactorType" type="text_en" multiValued="true" stored="true" indexed="true"/>
     <field name="subject" type="text_en" multiValued="true" stored="true" indexed="true"/>
     <field name="subtitle" type="text_en" multiValued="false" stored="true" indexed="true"/>
@@ -613,13 +613,13 @@
     <copyField source="studyAssayOrganism" dest="_text_" maxChars="3000"/>
     <copyField source="studyAssayOtherMeasurmentType" dest="_text_" maxChars="3000"/>
     <copyField source="studyAssayOtherOrganism" dest="_text_" maxChars="3000"/>
-    <copyField source="studyAssayPlatform" dest="_text_" maxChars="3000"/>
     <copyField source="studyAssayOtherPlatform" dest="_text_" maxChars="3000"/>
-    <copyField source="studyAssayTechnologyType" dest="_text_" maxChars="3000"/>
     <copyField source="studyAssayOtherTechnologyType" dest="_text_" maxChars="3000"/>
+    <copyField source="studyAssayPlatform" dest="_text_" maxChars="3000"/>
+    <copyField source="studyAssayTechnologyType" dest="_text_" maxChars="3000"/>
     <copyField source="studyDesignType" dest="_text_" maxChars="3000"/>
-    <copyField source="studyOtherDesignType" dest="_text_" maxChars="3000"/>
     <copyField source="studyFactorType" dest="_text_" maxChars="3000"/>
+    <copyField source="studyOtherDesignType" dest="_text_" maxChars="3000"/>
     <copyField source="studyOtherFactorType" dest="_text_" maxChars="3000"/>
     <copyField source="subject" dest="_text_" maxChars="3000"/>
     <copyField source="subtitle" dest="_text_" maxChars="3000"/>
```

File: conf/solr/8.8.1/schema.xml (hunk 1 above)
Comment (pdurbin) [reply conf/solr/8.8.1/schema.xml:362]: I added the subfields of series in 4509dea.

**Comment by pdurbin** [comment]
In 4509dea I updated schema.xml because the subfields of "series" had not been made multivalued. In #8248 we plan to explain the recommended way for developers to update schema.xml when making changes to metadata blocks we ship with the app, such as citation. In short, you should use the output from http://localhost:8080/api/admin/index/solr/schema

I've been poking a Solr and search-related stuff on this pull request. Somewhat surprisingly, it seems like a full reindex may not be required. I was able to create a dataset under "develop" and still have it's newly multivalued fields show up as facets once I deployed the new code.

Once I had the new code deployed, however, I noticed some bugs. Here's what I posted in Slack:

DDI export fails (and maybe other exports?) if you specify multiple production places.

“More than one series title found” is logged over and more if you specify more than series. It seems to be used in the citation.

Basically, there is hard-coded logic around these fields in the Java code. Changing the TSV is not enough.

A rule of thumb might be… if you’re messing with citation fields, go in DatasetFieldConstant and do a “find usages” for the fields you’re changing and try to figure out the impact.

**Comment by djbrooke** [comment]
Hi @philippconzett, as you can see in the note from @pdurbin, during review and testing we realized that these changes will take more work to implement than just updates to the .tsv and accompanying release notes and Solr schema changes. I'll need to look at our other work in progress to see if this is something that we can prioritize, but I don't believe we'll be able to work on the necessary code changes here before the end of the year.

File: conf/solr/8.8.1/schema.xml (hunk 1 above)
Comment (philippconzett) [reply conf/solr/8.8.1/schema.xml:362]: Or wouldn't it make more sense to change Series to a complex field, such as Author? I guess this would mean:

series
allowmultiples: TRUE

seriesName
allowmultiples: FALSE

seriesInformation
allowmultiples: FALSE

**Comment by pdurbin** [comment]
I just wanted to mention that we recently gained some experience changing allowMultiples from false to true with "Collection Mode" (collectionMode) in PR #8473. One thing to look out for is to make sure DDI import continues to work. See PR #8491 and PR #8483.

**Comment by jggautier** [comment]
This applies the change at the installation level, so it's not possible for managers of collections within a Dataverse installation to control the number of values allowed for a field. Are there collection managers within Dataverse repositories who want to make sure that depositors can enter only Production Place?

Should the same question be investigated for other fields we consider making repeatable, including Collection Mode (although that field's already been made repeatable), before solutions are proposed and implemented?

**Comment by pdurbin** [comment]
@philippconzett heads up that there are merge conflicts in this PR. Do you need a hand resolving them?

**Comment by pdurbin** [comment]
@philippconzett and I just had a call about this PR and we're closing it in favor of this one:

- #9255
